### DISCUSSION TITLE
Memory leak?

### DISCUSSION THREAD

**Discussion by gwww** [python-pillow/Pillow#8166] [General]
I suspect a memory leak in the following code. Reproducing it will take a bit of effort to decouple the code below from a bigger library, so before I try to reproduce, is the code considered a good practice or is there a better different way to generate a GIF from a set of PNGs?

`radar_layers` contains a set of about ~20 PNG images.

The snippet below comes from https://github.com/gwww/env_canada/blob/new-radar/env_canada/ec_radar.py#L352-L363. I'm investigating this issue: https://github.com/home-assistant/core/issues/109622

From the larger piece of code I've narrowed it down to the lines referenced as the rest of the code is just reading cached image bytes after the first run and `create_gif` is close to the only thing left.

I have read a number of issues in this repo (both open and closed) searching on "memory leak". Nothing I saw gave me any hints, but I did not read them all.

```python
        # FYI - import imageio.v2 as imageio
        def create_gif():
            """Assemble animated GIF."""
            duration = 1000 / fps
            gif_frames = [imageio.imread(f, mode="RGBA") for f in radar_layers]
            gif_bytes = imageio.mimwrite(
                imageio.RETURN_BYTES,
                gif_frames,
                format="GIF",
                duration=duration,
                subrectangles=True,
            )
            return gif_bytes
```

**Comment by Yay295** [comment]
Can you reproduce the issue with Pillow directly rather than using imageio?

**Reply by gwww** [reply]
Thanks. I did recreate and isolated to imageio. I'll continue discussion there.
